src/Microsoft.Health.Abstractions/Data/ISource.cs

```diff
@@ -0,0 +1,27 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Health.Abstractions.Data
+{
+    /// <summary>
+    /// Used as the source for Data of type T.
+    /// </summary>
+    /// <typeparam name="T">Records of Type T</typeparam>
+    public interface ISource<T>
```

Comment:
Same as above, we could be less general and more specific here. Maybe IEventSource ?